crates/api_crud/src/site/update.rs

```diff
@@ -208,3 +183,201 @@ impl PerformCrud for EditSite {
     Ok(res)
   }
 }
+
+fn validate_update_payload(
+  site_regex: Option<String>,
+  federation_enabled: bool,
+  private_instance: bool,
+  edit_site: &EditSite,
+) -> LemmyResult<()> {
+  // Check that the slur regex compiles, and return the regex if valid...
+  let slur_regex = build_and_check_regex(&site_regex.as_deref())?;
+
+  if let Some(name) = &edit_site.name {
+    // The name doesn't need to be updated, but if provided it cannot be blanked out...
+    site_name_length_check(name)?;
+    check_slurs_opt(&edit_site.name, &slur_regex)?;
+  }
+
+  if let Some(desc) = &edit_site.description {
+    site_description_length_check(desc)?;
+    check_slurs_opt(&edit_site.description, &slur_regex)?;
+  }
+
+  if let Some(listing_type) = &edit_site.default_post_listing_type {
```

Comment:
Should this check also be done when creating a site? Same for the federation vs private checks below. I can put these into a function in `validation.rs` and re-use them for the creation API.

Reply:
Its probably a good idea, for consistency, although lemmy-ui doesn't show nearly as many fields when creating the site, as for updating it.